modules/external_petsc_solver/src/petscsolver/PETScDiffusionFDM.C

```diff
@@ -147,6 +148,12 @@ externalPETScDiffusionFDMSolve(TS ts, Vec u, PetscReal dt, PetscReal time)
      - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - */
   ierr = TSSolve(ts, u);
   CHKERRQ(ierr);
+  if (converged)
```

Comment:
This seems a little odd. The parameter is required, but can be explicitly set to `nullptr`. Is this what you want or do you want to make it an optional parameter? I guess there's nothing wrong with it. It just seems odd is all.

Reply:
In the first place, I was thinking we could just leave that parameter as optional. Sometimes, I do not need to check the convergence status. It may be a good idea to check it all the time. Then make the parameter required.